tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -87,6 +168,19 @@ void unordered_containers_test(
     assert(something != different);
 }
 
+template <class ErrorType>
+void test_diagnostics_type() {
+    dummy_diagnostic* c_mem = new dummy_diagnostic[2];
+
+    ErrorType e_smaller(0, c_mem[0]);
+    ErrorType e_equal(0, c_mem[0]);
+    ErrorType e_larger(1, c_mem[1]);

```

Comment:
Is this exercising all of the runtime codepaths? I expected to see test cases for "categories are different" and "categories are same; tiebreak on values".